pkg/cli/connect.go

```diff
@@ -0,0 +1,189 @@
+package cli
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/QMSTR/qmstr/pkg/service"
+	"github.com/spf13/cobra"
+	"golang.org/x/net/context"
+)
+
+var validFileToFileEdges = []string{
+	"derivedFrom",
+	"dependencies",
+}
+
+var validFileToPackageEdges = []string{
+	"targets",
+}
+
+var connectCmdFlags = struct {
+	fileToFileEdge    string
+	fileToPackageEdge string
+}{}
+
+func init() {
+	rootCmd.AddCommand(connectCmd)
+	connectCmd.Flags().StringVar(&connectCmdFlags.fileToFileEdge, "fileToFileEdge",
+		"derivedFrom", fmt.Sprintf("Edge to use when connecting FileNode to FileNode. One of %v", validFileToFileEdges))
+	connectCmd.Flags().StringVar(&connectCmdFlags.fileToPackageEdge, "fileToPackageEdge",
+		"targets", fmt.Sprintf("Edge to use when connecting FileNode to PackageNode. One of %v", validFileToPackageEdges))
+}
+
+var connectCmd = &cobra.Command{
+	Use:   "connect",
+	Short: "Connect nodes with specific edges",
+	Long: `Usage: qmstrctl connect <that> <this>...
+Connect to Node <that> Node(s) <this>. In case of multiple edges for the specified types you can use --<type>To<type>Edge flag to specify the edge you want.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		setUpControlService()
+		setUpBuildService()
+		if err := connectCmdRun(cmd, args); err != nil {
```

Comment:
You are not using the cmd in connectCmdRun(cmd, args). Better omit that.

Reply:
The function is consistent with cobra's signature. It doesn't matter if i am using it now or not.